pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -373,3 +425,71 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     return nd_shape;
 
 }
+
+void array_copy_data(t_ndarray dest, t_ndarray src, int64_t pad)
+{
+    int64_t index = 0;
+    int64_t shape_product_src = 1;
+    int64_t shape_product_dest = 1;
+    int64_t var = 0;
+    while (index < dest.length)
+    {
+        int64_t ndim_src = src.nd - 1;
+        int64_t ndim_dest = dest.nd - 1;
+        int64_t i_src = (index % src.shape[ndim_src]) * src.strides[ndim_src];
+        int64_t i_dest = (index % dest.shape[ndim_dest]) * dest.strides[ndim_dest];
+        ndim_src--;
+        ndim_dest--;
+        while (ndim_src >= 0)
+        {
+            // i_src calculation

```

Comment:
It represents the next elements in an ndarray, depending on the ndarray's shape and strides, for example, if the order is F, we'll need to take the strides into consideration while parkouring it, since the elements have a Fortran memory layout, same goes for sliced arrays/views.
I got this function from a merge with @soukainahikma, since her method was way better than the stuff I was doing, I didn't get a full grasp of how the function works, I'll check it again.

Reply:
Ok. Maybe @soukainahikma can suggest a more explicit name. If you can't think of anything clear and compact then please add a comment